### PR TITLE
Fix new Coverity warnings

### PR DESCRIPTION
**Pull request by lmoureaux**
A handful of Coverity warnings popped up in the last month. This should clear them all.

Special thanks to Synopsis for 55336e7b7bc37cb046c78a259fa6a99dbb985266: *The client was crashing when Ctrl+Alt+clicking outside of the map (this shortcut is bound to the "create link" action).*

### CODE REVIEW

File: server/auth.cpp [hunk 1]
```diff
@@ -35,11 +35,6 @@ _   ._       Copyright (c) 1996-2021 Freeciv21 and Freeciv contributors.
 #define GUEST_NAME "guest"
 
 #define MIN_PASSWORD_LEN 6 // minimum length of password
```
Comment (pranavtifr) [inline server/auth.cpp:37]: Cant we remove MIN_PASSWORD_LEN and is_good_password() entirely (just check for ascii and garbage and keep the function as is_password_valid() or something) ? If we dont want to have custom servers for password quality, we shouldnt bother with length too imo.

File: tools/civmanual.cpp [hunk 2]
```diff
@@ -644,6 +644,7 @@ static bool manual_command(struct tag_types *tag_info)
       break;
 
     case MANUAL_COUNT:
```
Comment (pranavtifr) [inline tools/civmanual.cpp:646]: Why do we need to keep this MANUAL_COUNT again ? Some backward compatibility ? If yes, we need to comment it.

**Comment by pranavtifr** [comment]
The fails dont look like garbage, WA and Win doesnt seem to like the `std::array` conversion.

File: tools/civmanual.cpp (hunk 2 above)
Comment (lmoureaux) [reply tools/civmanual.cpp:646]: It's the upper bound for the loop iterating over all manual types.

File: server/auth.cpp (hunk 1 above)
Comment (lmoureaux) [reply server/auth.cpp:37]: Let's say that such a discussion is out of the scope of this PR which is to fix Coverity warnings about always-true conditionals. We can open an issue if we want to remove it.

**Comment by pranavtifr** [comment]
All the checks have passed now. Merging.

**Pull request merged**